src/vendor/azure-uamqp-c/src/message.c

```diff
@@ -1447,3 +1465,71 @@ int message_get_message_format(MESSAGE_HANDLE message, uint32_t *message_format)
 
     return result;
 }
+
+int message_set_delivery_tag(MESSAGE_HANDLE message, AMQP_VALUE delivery_tag_value)
+{
+    int result;
+
+    if ((message == NULL) || (delivery_tag_value == NULL))
+    {
+        LogError("Bad arguments: message = %p, delivery_tag = %p",
+            message, delivery_tag_value);
+        result = __FAILURE__;
+    }
+    else
+    {
+        AMQP_VALUE new_delivery_tag = amqpvalue_clone(delivery_tag_value);
+        if (new_delivery_tag == NULL)
+        {
+            LogError("Cannot clone delivery tag");
+            result = __FAILURE__;
+        }
+        else
+        {
+
+            message->delivery_tag = new_delivery_tag;
+
+            /* Codes_SRS_MESSAGE_01_102: [ On success it shall return 0. ]*/
+            result = 0;
+        }
+    }
+
+    return result;
+}
+
+int message_get_delivery_tag(MESSAGE_HANDLE message, AMQP_VALUE *delivery_tag_value)
+{
+    int result;
+
+    if ((message == NULL) ||
+        (delivery_tag_value == NULL))
+    {
+        LogError("Bad arguments: message = %p, delivery_tag = %p",
+            message, delivery_tag_value);
+        result = __FAILURE__;
+    }
+    else
+    {
+        if (message->delivery_tag == NULL)
+        {
+            *delivery_tag_value = NULL;
+            result = 0;
+        }
+        else
+        {
+            AMQP_VALUE new_delivery_tag = amqpvalue_clone(message->delivery_tag);
```

Comment:
we can just do it this way without declaring the `new_delivery_tag`?
```c
*delivery_tag_value = amqpvalue_clone(message->delivery_tag);
if(*delivery_tag_value == NULL)
{
    LogError("Cannot clone delivery tag");
    result = __FAILURE__;
}
else
{
    return = 0;
}
```

Reply:
This may be a silly reason for me to feel unsure about that suggestion, but causing a side effect (setting the output value to null) in the failure case seems a little bit "impure" given that we're communicating failure via the return variable, at least in terms of saving effectively a line of code.  If there's precedence in other functions that overrides this feel free to entirely ignore me.

Reply:
@KieranBrantnerMagee you're right
no problem for this one